### PR TITLE
test: bump CSI container versions for mock testing

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

These are the latest stable releases. We should test with those.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ spec:
             - mountPath: /csi
               name: socket-dir
         - name: driver-registrar
-          image: k8s.gcr.io/sig-storage/csi-node-driver-registrar:v1.3.0
+          image: k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
```
Comment (pacoxu) [inline test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver.yaml:32]: https://github.com/kubernetes/k8s.io/blob/798f7d6ac63d50dc79e96f5915644fad1f2e851b/k8s.gcr.io/images/k8s-staging-sig-storage/images.yaml#L15
I checked the image list here and find that v2.1.0 is the latest one.

Comment (pohly) [reply test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver.yaml:32]: Darn, a typo. I was looking at the list of images at https://console.cloud.google.com/gcr/images/k8s-artifacts-prod/US/sig-storage/csi-node-driver-registrar?gcrImageListsize=30 and then swapped digits. :cry:

Comment (pohly) [reply test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver.yaml]: Thanks for checking. Fixed now.

**Comment by pohly** [comment]
Hmm, looks like this needs further work: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/98770/pull-kubernetes-e2e-gce-storage-slow/1357607288189227008/

**Comment by pohly** [comment]
The newer sidecars needed some command line parameter changes. Should be fixed now.

**Review by pacoxu — APPROVED** [review]
/lgtm

**Comment by pohly** [comment]
/assign @xing-yang

For approval.

**Pull request merged**